gobblin-core/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -88,27 +82,6 @@
   public static final String AVG_RECORD_SIZE = "avg.record.size";
   public static final String AVG_RECORD_MILLIS = "avg.record.millis";
 
-  /**
-   * A configuration key that allows a user to specify config parameters on a topic specific level. The value of this
-   * config should be a JSON array. Each entry should be a {@link JsonObject} and should contain a
-   * {@link com.google.gson.JsonPrimitive} that identifies the {@link #TOPIC_NAME}. All configs in each topic entry will
-   * be added to the {@link WorkUnit}s for that topic.
-   *
-   * <p>
-   *   An example value could be: "[{"topic.name" : "myTopic1", "writer.partition.columns" : "header.memberId"},
-   *   {"topic.name" : "myTopic2", "writer.partition.columns" : "auditHeader.time"}]".
-   * </p>
-   *
-   * <p>
-   *   The "topic.name" field also allows regular expressions. For example, one can specify key, value
-   *   "topic.name" : "myTopic.*". In this case all topics whose name matches the pattern "myTopic.*" will have all the
-   *   specified config properties added to their {@link WorkUnit}s. If more than one topic matches multiple "topic.name"s then
-   *   the properties from all the {@link JsonObject}s will be added to their {@link WorkUnit}s.
-   * </p>
-   */
-  @VisibleForTesting
-  static final String KAFKA_TOPIC_SPECIFIC_STATE = "kafka.topic.specific.state";
```

Comment:
will this property still work with the new approach for adding topic specific state in the `QueryBasedSource`? Since Kafka ingestion is a pretty popular use case, we should maintain backwards compatibility. I believe I have seen at least one person on the google groups refer to using this config.